src/Microsoft.NET.Sdk.Functions.Generator/FunctionJsonConverter.cs

```diff
@@ -129,8 +123,7 @@ public IEnumerable<(FunctionJsonSchema schema, FileInfo outputFile)?> GenerateFu
                             var path = Path.Combine(_outputPath, artifactName);
                             var relativeAssemblyPath = PathUtility.MakeRelativePath(Path.Combine(_outputPath, "dummyFunctionName"), type.Assembly.Location);
                             var functionJson = method.ToFunctionJson(relativeAssemblyPath);
-                            if (CheckAppSettingsAndFunctionName(functionJson, method) &&
-                                _buildArtifactsLog.TryAddBuildArtifact(artifactName))
+                            if (CheckAppSettingsAndFunctionName(functionJson, method))
```

Comment:
As Ahmed was mentioning, can we change this to look for a new attribute (Ignore) and skip the generation if present? Wouldn't this be a much simpler change?

Reply:
So after I implemented the attribute approach, I think this one is actually better. 
The reason is that if the user has a folder in their project that contains a `function.json` that's unrelated and is set to copy to output, the build task would delete it with the attribute approach because we have to clear all folders containing `function.json` in case a function was renamed or removed between compilations. 

The process that I had is 
1. get all functions with [SkipAutoGen]
2. delete all folders containing function.json from the output **except** those in the list above
3. generate new function.json files

if we don't do (2) that will cause deleted functions or renamed functions to leave stuff behind. but (2) would also delete any folders with function.json in them that we don't know anything about. I think this is a corner case so I don't know how important that is. This approach however avoids this issue.

Reply:
if the user has a function.json that needs to be used, shouldn't it be as easy as creating a class with function name and new attribute. I think the attribute approach is a lot cleaner since the tooling support is for pre-compiled functions.

Reply:
I think the attribute based approach would be simpler. The reason why I have chosen the proj file based approach was compatibility with existing projects. The usage of an attribute would force users to change their implementation (if they provide custom `function.json` files), whereas the proj based approach should work with existing projects without change (except in some corner cases maybe).

Two other approaches:
1. Using a naming convention. A user could place the `function.json` files in subfolders of a folder named `FunctionDescriptions` (or some better name) and all subfolders would be copied to the output folder (after deleting all `function.json` files and (re)generating). But this approach would not respect the proj file settings (as the attribute approach would neither).

2. Could we identify the function.json files that were generated by the generator by looking at the generated file? There is a property `generatedBy`. Could we use this property to identify files that were generated by the generator (or to identify user provided files)?

Reply:
since this approach works and is backward compatible, I like it more than other options. I'll merge this PR. If in the future we still need the [SkipAutoGen] attribute in dcfa8332b0e249e83c77dad548d0c1fe04965ae7